src/coreclr/src/vm/processdiagnosticsprotocolhelper.cpp

```diff
@@ -47,6 +47,7 @@ uint16_t ProcessInfoPayload::GetSize()
     // LPCWSTR CommandLine; -> 4 bytes + strlen * sizeof(WCHAR)
     // LPCWSTR OS;          -> 4 bytes + strlen * sizeof(WCHAR)
     // LPCWSTR Arch;        -> 4 bytes + strlen * sizeof(WCHAR)
+    // Array<LPCWSTR> Environment; -> 4 bytes + total bytes of strings
```

Comment:
Do you have any sense of how much more expensive in terms of time spent handling this command with the environment block changes vs without? I'm wondering if it makes sense to have getting the environment block as a separate command so that consumers who want basic information can do so without getting the environment block and then they can decide to get the environment block later if need be.

Reply:
For example, I plan on having dotnet-monitor get and store the basic information (pid, runtime instance identifier, command line, etc) once when the runtime connects to the diagnostic port listener of dotnet-monitor (which requires issuing the GetProcessInfo command) and then issuing the same command later on demand to get the environment block when displaying the list of available processes (because the environment block can change over time). Maybe in the grand scheme of how dotnet-monitor works, having the environment block reported via the GetProcessInfo command doesn't really have much of a time perf impact, but it might be something to think about (splitting it into its own command).